### ISSUE TITLE
Problem

### ISSUE THREAD

**Issue by SuperUser69420** [PiiluDuck/SuperPNGPhotoshop#1]
Hey this is exactly what I been looking for! But I ran into an issue, I'm pretty noob when it comes to coding and for some reason it wont export correctly? Can you explain to me how to set it up properly for it to export correctly, Is there a special way for it to work? Does this export the layers to files? Also is there a way to change the quality of export? So far what I did was changed the file scripts name with .jsx to the end and put them in my script folder in photoshop and ran it with a few layers but after selecting the folder i don't get any exported superPNG files? Is there something I am doing wrong?

**Comment by PiiluDuck** [comment]
Sorry for not replaying quicker. Basically there are two options. One with transparency and one without. It uses best settings for PNG for texture work when using Photoshop and SuperPNG. If you wanna use your own settings it gets quite tricky, as SuperPNG uses its own weird parameters only way to know what and how to set them is by using something that records your actions in photoshop and outputs them in Java. There is an app or API or something but I forgot the name. If you find that tool on google you can record your actions and then select and paste the parameters in the code.

**Comment by PiiluDuck** [comment]
> So far what I did was changed the file scripts name with .jsx to the end and put them in my script folder in photoshop and ran it with a few layers but after selecting the folder i don't get any exported superPNG files? Is there something I am doing wrong?

Ou I think I see the problem. Just open your favorite code writing program, be it Visual or note++, its Java code, so it really doesent matter. Select the version of the code you wanna use, copy and paste it into the file and save it with the extension _.js_ save. Then open photoshop, navigate to File - Script - Browse..., find and select the script you just saved and done. You can now mass convert files.

**Comment by SuperUser69420** [comment]
I'm not exactly sure what I am doing wrong exactly.. but for some reason it wont work properly? Is it supposed to export photoshop layers to superPNG files? That's what I am trying to do. When I do run your script the only way it exports anything is if i put some already png files in the folder i select but it doesn't properly export still it only will do like 1 layer export like that. I pretty much need this type of thing here https://graphicdesign.stackexchange.com/questions/119744/photoshop-export-layers-to-files-using-plugin but when I ran those scripts on that URL i get so many errors and it doesn't want to run sadly. But with your script i get no errors but it doesnt export the layers for me. And yes i tried the actions to scripting thing and when i try to run a script from what is created it gives me errors it's super weird. I just want to mass export layers to superPNG format, I'm not sure why yours isn't working properly unless I'm supposed to set up a folder in a special way or something? Is there anyway you could maybe make a video clip of how you run the script and what happens? I'm on Windows 10 and tried the script on Photoshop CC 2019 and Photoshop 2022 btw

**Comment by PiiluDuck** [comment]
@SuperUser69420 the script is designed to quickly save files from a folder. Let's say that you have a folder of standard PNG files, well with this script you can convert them into super PNG compression format. It won't save layers sadly or export them that way. This script is just a mass converter :/.

**Comment by SuperUser69420** [comment]
Ohhh, Is there anyway you could alter this script to do photoshop layers exported into superPNG file compression format? I pretty much need the standard feature photoshop has under File > Export > Layers to Files... but with the superPNG format, I been looking for a script to do this for over a year now and can't find one that works :(

**Comment by PiiluDuck** [comment]
Sadly I think it's not possible... Well it is, but no point really. Best way is to export with the standard PNG file and then use the script to format all to super PNG. That's for what the script is made for. To convert a folder of standard PNG files into super PNG compression. :)

**Comment by SuperUser69420** [comment]
Yeah the only thing I dislike about that is the standard PNG File exporting feature on photoshop is sooo slow for me when I do 50+ layers.

**Comment by SuperUser69420** [comment]
I just tested it got it working, It will definitely work for me for now until I can find or figure out how to do layers to superPNG. Also I wanted to let you know that the names of each script is backwards, The no_trans is actually the transparent one and the with_trans one is the no transparent background. I do appreciate the help and replies btw thank you! :)
